src/components/gltf-model-plus.js

```diff
@@ -341,127 +340,221 @@ function runMigration(version, json) {
+    }
+    runMigration(version, parser.json);
 
-  const parser = await new Promise((resolve, reject) => gltfLoader.createParser(gltfUrl, resolve, onProgress, reject));
+    // Note: Here may be rewritten with the one with parser.associations
+    const nodes = parser.json.nodes;
+    if (nodes) {
+      for (let i = 0; i < nodes.length; i++) {
+        const node = nodes[i];
 
-  parser.textureLoader = new HubsTextureLoader(loadingManager);
+        if (!node.extras) {
+          node.extras = {};
+        }
 
-  if (jsonPreprocessor) {
-    parser.json = jsonPreprocessor(parser.json);
+        node.extras.gltfIndex = i;
+      }
+    }
   }
 
-  let version = 0;
-  if (
-    parser.json.extensions &&
-    parser.json.extensions.MOZ_hubs_components &&
-    parser.json.extensions.MOZ_hubs_components.hasOwnProperty("version")
-  ) {
-    version = parser.json.extensions.MOZ_hubs_components.version;
-  }
-  runMigration(version, parser.json);
+  afterRoot(gltf) {
+    gltf.scene.traverse(object => {
+      // GLTFLoader sets matrixAutoUpdate on animated objects, we want to keep the defaults
+      // @TODO: Should this be fixed in the gltf loader?
+      object.matrixAutoUpdate = THREE.Object3D.DefaultMatrixAutoUpdate;
+      const materialQuality = window.APP.store.materialQualitySetting;
+      object.material = mapMaterials(object, material => convertStandardMaterial(material, materialQuality));
+    });
 
-  const nodes = parser.json.nodes;
-  if (nodes) {
-    for (let i = 0; i < nodes.length; i++) {
-      const node = nodes[i];
+    //
+    gltf.scene.traverse(object => {
+      if (!object.material) {
+        return;
+      }
+      const materials = Array.isArray(object.material) ? object.material : [object.material];
+      // @TODO: Handle more efficiently
+      for (const material of materials) {
+        for (const key in material) {
+          const prop = material[key];
+          if (prop && prop.isTexture && !prop.onUpdate) {
+            prop.onUpdate = () => {
+              // Delete texture data once it has been uploaded to the GPU
+              prop.image.close && prop.image.close();
+              delete prop.image;
+            };
+          }
+        }
+      }
+    });
 
-      if (!node.extras) {
-        node.extras = {};
+    // Replace animation target node name with the node uuid.
+    // I assume track name is 'nodename.property'.
+    if (gltf.animations) {
+      for (const animation of gltf.animations) {
+        for (const track of animation.tracks) {
+          const parsedPath = THREE.PropertyBinding.parseTrackName(track.name);
+          for (const scene of gltf.scenes) {
+            const node = THREE.PropertyBinding.findNode(scene, parsedPath.nodeName);
+            if (node) {
+              track.name = track.name.replace(/^[^.]*\./, node.uuid + ".");
+              break;
+            }
+          }
+        }
       }
+    }
 
-      node.extras.gltfIndex = i;
+    //
+    if (this.fileMap) {
```

Comment:
Might be cleaner to just handle this in `loadGLTF` instead of passing `fileMap` through.

Reply:
Sounds good. Updated.